pkg/astrolabe/protected_entity.go

```diff
@@ -200,4 +200,10 @@ type ProtectedEntity interface {
 	// best data path to provide the Reader stream.  If the ProtectedEntity does not have any metadata, nil will be
 	// returned
 	GetMetadataReader(ctx context.Context) (io.ReadCloser, error)
+
+	// Overwrite replaces the data and metadata of the ProtectedEntity with the metadata from the sourcePE
+	// The ID of the protected entity does not change
+	// Any components will also be overwritten from the components of the sourcePE if the overwriteComponents flag is set
+	Overwrite(ctx context.Context, sourcePE ProtectedEntity, params map[string]map[string]interface{},
```

Comment:
The Overwrite API looks more like a data path API rather than a control path one. Currently, Astrolabe APIs are pretty much organized as that, control path APIs, like Snapshot, DeleteSnaphsot, etc, are placed in ProtectedEntity interface, while the data path APIs, like Copy, CopyFromInfo, are placed in ProtectedEntityTypeManager interface. I am curious about why not place it together with Copy API in PETM interface. 

Additionally, if we place it along side with the Copy API in PETM interface, we can find that Copy and Overwrite could be very similar.  
```
Copy(ctx context.Context, srcPE ProtectedEntity, dstPE ProtectedEntity, params map[string]map[string]interface{}, options CopyCreateOptions) (ProtectedEntity, error)
```
Do we consider merging them into a single API? The good part is we don't have to add a new API. But, the downside is, the implementation of the Copy API will end up to be more complicated.

Reply:
Currently, the Copy APIs either create a new object or overwrite an object with the same ID as the source object.  That's why they wound up in PETM, not a desire to split control/data.  So the PETM APIs are things that are across multiple objects or create new objects.  I added the Overwrite to the PE because it is an operation on an existing object.